Formula/haskell-language-server.rb

```diff
@@ -20,29 +21,43 @@ class HaskellLanguageServer < Formula
   end
 
   depends_on "cabal-install" => [:build, :test]
-  depends_on "ghc" => [:build, :test]
-  depends_on "ghc@8.6" => [:build, :test]
-  depends_on "ghc@8.8" => [:build, :test]
+
+  on_macos do
+    depends_on "ghc" => [:build, :test]
+    depends_on "llvm" => :build if Hardware::CPU.arm?
+
+    unless Hardware::CPU.arm?
+      depends_on "ghc@8.6" => [:build, :test]
+      depends_on "ghc@8.8" => [:build, :test]
+    end
+  end
+
+  on_linux do
+    depends_on "ghc" => [:build, :test]
+    depends_on "ghc@8.6" => [:build, :test]
+    depends_on "ghc@8.8" => [:build, :test]
+  end
```

Comment:
Do we need a separate linux section anymore?
Based on other Formulae, I don't think we usually care about Linux ARM as a use-case.
```rb
  depends_on "ghc" => [:build, :test]
  depends_on "llvm" => :build if Hardware::CPU.arm?
  unless Hardware::CPU.arm?
    depends_on "ghc@8.6" => [:build, :test]
    depends_on "ghc@8.8" => [:build, :test]
  end
```